ui/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ManagementActor.scala

```diff
@@ -116,9 +117,16 @@ class ManagementActor(environment: String,
       reply(Future.successful(DeploymentStatusResponse(beingDeployed)))
   }
 
+  private def handleCheckStatus(processState: Option[ProcessState], deployedVersion: Option[Long], lastDeployHistory: Option[DeploymentHistoryEntry]): Option[ProcessStatus] =
```

Comment:
What does mean this method name? You handle some checking? What checking? Maybe `handleObsoleteManagerState`? This code is very Flink-hermetic, we should add very verbose comment here e.g.:
```
Some ProcessManagers (e.g. Flink's) lost information about canceled state after some duration. In case of this we use state from our cached history of states. TODO: we should move this ManagementActor behind ProcessManager API and do this logic only in Flink's case.
```

Reply:
Done.